flash-patcher.py

```diff
@@ -24,111 +15,147 @@
+            stdout=subprocess.DEVNULL,
+            check=True,
+        )
+
+    def export_scripts(
+        self: JPEXSInterface,
+        inputfile: Path,
+        output_dir: Path,
+    ) -> subprocess.CompletedProcess:
+        """Export scripts from a SWF file into a directory."""
+        info("Exporting scripts into %s...", output_dir)
+        return subprocess.run(
+            [
+                self.path,
+                *self.args,
+                "-export",
+                "script",
+                output_dir,
+                inputfile,
+            ],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+            check=True,
+        )
+
+    def recompile_data(
+        self: JPEXSInterface,
+        part: str,
+        swf: Path,
+        output: Path,
+    ) -> int:
+        """Recompile data of a given type into the SWF file."""
+        info("Reimporting %ss...", part)
+        return subprocess.run(
+            [
+                self.path,
+                *self.args,
+                f"-import{part}",
+                swf,
+                output,
+                DECOMP_LOCATION,
+            ],
+            stdout=subprocess.DEVNULL,
+            check=True,
+        )
 
-"""
-Find the location in the file specified.
-If code is an integer, it'll resolve to writing AFTER that line number.
-If code is "end", it'll resolve to the end of that file.
-"""
-def find_write_location(lines, code):
-    if (code == "end"):
-        return len(lines)
-    else:
-        try:
-            return int(code) - 1
-        
-        except ValueError:
-            perror("")
-            perror("Invalid add location: " + code)
-            perror("Expected keyword or integer (got type \"str\")")
-            perror("Aborting...")
-            exit(1)
 
 class FilePosition:
-    def __init__(self, file_name):
+    """A position in a named file."""
+
+    def __init__(self: FilePosition, file_name: str) -> None:
         self.fileName = file_name
         self.lineNumber = 0
 
+
 class CodeInjector:
-    def __init__(self):
+    """Handles injection of code into script files."""
+
+    def __init__(self: CodeInjector) -> None:
         self.files = []
         self.fileContents = {}
         self.injectLines = []
         self.startingLineNo = -1
 
     # Return the file name of the script being modified
-    def addInjectionTarget(self, injection_info, patch_file, current_line_no):
+    def add_injection_target(
+        self: CodeInjector,
+        injection_info: list,
+        patch_file: Path,
+        current_line_no: int,
+    ) -> str:
+        """Add an injection target to this injector."""
         split_line = injection_info.split()
-        short_name = ' '.join(split_line[1:-1])
-        file_name = DECOMP_LOCATION_WITH_SCRIPTS + short_name
+        short_name = " ".join(split_line[1:-1])
+        file_name = DECOMP_LOCATION_WITH_SCRIPTS / short_name
```

Comment:
What does `/` do with strings?

Reply:
`DECOMP_LOCATION_WITH_SCRIPTS` isn't a string, it's a `Path` (from `pathlib`). `/` performs an append operation. This might be a mistake, though - I'm not quite sure how `short_name` is supposed to behave.

Reply:
Should be fine, actually.

Reply:
`short_name` is the path from inside the script folder, so this works fine.